pkg/controller/cache/managed.go

```diff
@@ -164,8 +164,6 @@ func (e *external) Create(ctx context.Context, mg resource.Managed) (resource.Ex
 	if !ok {
 		return resource.ExternalCreation{}, errors.New(errNotReplicationGroup)
 	}
-
-	cr.Status.SetConditions(runtimev1alpha1.Creating())
```

Comment:
Was this deletion intended?

Reply:
Yes. In other resources, this condition is set in `Observe` right after the object is retrieved, which I think makes sense to decide on the state with the observation we get rather than the action we're about to take. As an example, we can set condition to `Deleting` but the deletion call fails, however we'have marked the object as deleting even though it's perfectly available. This doesn't exactly map to `Creating` since there is no object yet but I like the idea of having the API's response as the only source of truth for the status.

Reply:
> In other resources, this condition is set in Observe right after the object is retrieved

I did not notice that, but I see now that it's true for all the CloudMemorystore controller, which will also need updating if we want to go this direction for v1beta1. In most (all?) pre-v1beta1 controllers the conditions are set in observe _and_ in create and delete.

That said, I'm not quite sold that we should go this direction. Frequently we describe the `Ready` condition as representing "the lifecycle of the managed resource", but I've always taken that to include capturing our controller's intent, rather than being based purely an observation of the external resource. Put otherwise, we _intentionally_ set the status to `Deleting` even if the delete call fails in order to communicate to the user that we received and are acting on their request to delete the managed resource.

Obviously this is a little confused by the fact that we updated statuses based on _actions_ in some places (create, delete), but on _observations_ in orders (observe), so in this example we'd set `Deleting`, then set `Available` during `Observe`, then set `Deleting` again when we tried to delete the resource again (due to the deletion timestamp being set). 

I do think there's some value in the current pattern. Given that we requeue every 30 seconds when waiting for an operation like create or delete this means when you create a `CloudSQLInstance` you won't see that it's creating until the first reconcile half a minute later.

Reply:
> Frequently we describe the Ready condition as representing "the lifecycle of the managed resource", but I've always taken that to include capturing our controller's intent, rather than being based purely an observation of the external resource.

I always think that status is simply the observation of the resource and this wouldn't include controller's intent. But I do see value in emitting the controller's intent.

> I do think there's some value in the current pattern. Given that we requeue every 30 seconds when waiting for an operation like create or delete this change means when you create a CloudSQLInstance you won't see that it's creating until the first reconcile half a minute later.

Good point.

I'll change all resources to set the condition in `Create/Delete`. Maybe we need a condition for `Updating`?

Reply:
@negz How about having managed reconciler do this condition setting before calling the corresponding functions in 0.5?

Reply:
Definitely a fan of adding this to the managed reconciler for consistency across controllers! 👍 I wonder if it would be useful to have some sort of controller status and provider status that are separate (i.e. "the state of the controller is Deleting but the state of the resource at the provider is Available" in the example above).

Reply:
@hasheddan well, actually `atProvider` has a state field that we calculate the availability from. It's just that the field name is not consistent across resources. In some resources it's `status.atProvider.dbInstanceStatus`, in some of them `status.atProvider.status` etc.

But I'm not sure if the drift between _state intent of the controller for the given resource_ and _state of the resource_ is worth a new condition. Because one converges to the other in max 1 minute in the happy path. For current cases;
* `Create` doesn't really make sense since there is no object if the intent isn't successfully executed,
* `Delete` makes sense to some extent but usually if `Delete` is requested not sure availability info is relevant anymore.

Apart from these, `Update` call is a bit different and that's where I see some value for separation. Because the stack controllers might want to take some actions in case of resource unavailability, like readiness reporting of the application could take managed resources' state into account and intermittent unavailability due to `Update` could be valuable to report. If the `Update` call isn't successful for some reason, stack controller should have the possibility to know the current exact state, which is resource is still safe to use.

Though currently we don't set a condition for `Update` calls, so, it should be fine. But I think it's worth noting that we should try to lean towards keeping the status sub-resource of the CR as close to actual status as possible for the sake of being a reliable lowest level of the Crossplane stacks.

Reply:
> Maybe we need a condition for Updating?

Yeah I've thought about this before and been on the fence. I agree it would especially make sense in cases where a resource was unavailable or somehow degraded while updating, but I'm less sure it makes sense if resources handle updates quickly and without degradation. Definitely worth opening an issue to think through.

> How about having managed reconciler do this condition setting before calling the corresponding functions in 0.5?

So the historical reason we ask folks to emit these conditions in the `ExternalClient` rather than the managed reconciler is because we didn't want to constrain managed resources to condition reasons like `Creating`, `Deleting`, and `Available`. e.g. Maybe a managed resource wants to name its "Creating" condition `Activating`, and its `Available` reason `Online` or similar. I lean slightly toward maintaining this flexibility, though I can see that we've taken advantage of it in zero cases so far.

> I wonder if it would be useful to have some sort of controller status and provider status that are separate

I tend to agree with @muvaf on this one. With our new `v1beta` resources in almost all cases you can see the exact status of the provider resource in its own language under `.status.atProvider`. In general I think we should be very conservative about adding new condition types.